sys/senml/phydat.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (C) 2021 Silke Hofstra
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdint.h>
+
+#include "math.h"
+#include "senml.h"
+#include "senml/phydat.h"
+
+
+
+static uint8_t phydat_unit_to_senml_unit(uint8_t unit)
+{
+    switch (unit) {
+    /* Compatible units */
+    case UNIT_TEMP_C:  return SENML_UNIT_CELSIUS;
+    case UNIT_TEMP_K:  return SENML_UNIT_KELVIN;
+    case UNIT_LUX:     return SENML_UNIT_LUX;
+    case UNIT_M:       return SENML_UNIT_METER;
+    case UNIT_M2:      return SENML_UNIT_SQUARE_METER;
+    case UNIT_M3:      return SENML_UNIT_CUBIC_METER;
+    case UNIT_GR:      return SENML_UNIT_GRAM;
+    case UNIT_A:       return SENML_UNIT_AMPERE;
+    case UNIT_V:       return SENML_UNIT_VOLT;
+    case UNIT_W:       return SENML_UNIT_WATT;
```

Comment:
```suggestion
    case UNIT_W:       return SENML_UNIT_WATT;
    case UNIT_T:       return SENML_UNIT_TESLA;
```

for when this is rebased after #17569